pkg/validation/validate.go

```diff
@@ -86,28 +89,76 @@ func ValidateIAMPolicyResource(ctx context.Context, pDoc v1alpha1.PolicyDocument
 	return nil
 }
 
-//ComparePolicy function compares input policy doc to target policy doc
-func ComparePolicy(ctx context.Context, request string, target string) bool {
+//CompareRole function compares input role to target role
+func CompareRole(ctx context.Context, request awsapi.IAMRoleRequest, role *iam.GetRoleOutput, target string) bool {
 	log := log.Logger(ctx, "pkg.validation", "ComparePolicy")
 
+	// Step 1: Compare the permission policy
+	if !ComparePermissionPolicy(ctx, request, target) {
+		return false
+	}
+
+	//Step 2: Compare Assume Role Policy Document
+	if !CompareAssumeRolePolicy(ctx, request, role) {
+		return false
+	}
+	//Step 3: Compare Permission Boundary
+	if !reflect.DeepEqual(request.ManagedPermissionBoundaryPolicy, *role.Role.PermissionsBoundary.PermissionsBoundaryArn) {
+		log.Info("input permission boundary and target permission boundary are NOT equal")
+		return false
+	}
+
+	return true
+}
+
+//ComparePermissionPolicy compares assume role policy from request and response
+func ComparePermissionPolicy(ctx context.Context, request awsapi.IAMRoleRequest, target string) bool {
```

Comment:
Similarly, ComparePermissionPolicy compares assume role policy from request and response. But, request (second parameter) is a roleRequest. Why don't we pass `request.PermissionPolicy` instead?